contracts/EVMScriptFactories/SetVettedValidatorsLimits.sol

```diff
@@ -0,0 +1,175 @@
+    /// @param _evmScriptCallData Encoded tuple: (uint256 _nodeOperatorId, uint256 _stakingLimit) where
+    /// _nodeOperatorId - id of node operator in NodeOperatorsRegistry
+    /// _stakingLimit - new staking limit
+    function createEVMScript(
+        address _creator,
+        bytes memory _evmScriptCallData
+    ) external view override onlyTrustedCaller(_creator) returns (bytes memory) {
+        VettedValidatorsLimitInput[] memory decodedCallData = _decodeEVMScriptCallData(
+            _evmScriptCallData
+        );
+
+        _validateInputData(decodedCallData);
+        
+        bytes[] memory setVettedValidatorsLimitsCalldata = new bytes[](decodedCallData.length);
+
+        for (uint i = 0; i < decodedCallData.length; i++) {
+            setVettedValidatorsLimitsCalldata[i] = abi.encode(
+                decodedCallData[i].nodeOperatorId,
+                decodedCallData[i].stakingLimit
+            );
+        }
+
+        return
+            EVMScriptCreator.createEVMScript(
+                address(nodeOperatorsRegistry),
+                nodeOperatorsRegistry.setNodeOperatorStakingLimit.selector,
+                setVettedValidatorsLimitsCalldata
+            );
+    }
+
+    /// @notice Decodes call data used by createEVMScript method
+    /// @param _evmScriptCallData Encoded tuple: (uint256 _nodeOperatorId, uint256 _stakingLimit) where
+    /// _nodeOperatorId - id of node operator in NodeOperatorsRegistry
+    /// _stakingLimit - new staking limit
+    /// @return VettedValidatorsLimitInput
+    function decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) external pure returns (VettedValidatorsLimitInput[] memory) {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) private pure returns (VettedValidatorsLimitInput[] memory) {
+        return abi.decode(_evmScriptCallData, (VettedValidatorsLimitInput[]));
+    }
+
+    function _validateInputData(VettedValidatorsLimitInput[] memory _decodedCallData) private view {
+        uint256 nodeOperatorsCount = nodeOperatorsRegistry.getNodeOperatorsCount();
+        for (uint i = 0; i < _decodedCallData.length; i++) {
+            require(
+                _decodedCallData[i].nodeOperatorId < nodeOperatorsCount,
+                NODE_OPERATOR_INDEX_OUT_OF_RANGE
+            );
+
+            NodeOperatorData memory nodeOperatorData = _getNodeOperatorData(
+                _decodedCallData[i].nodeOperatorId
+            );
+
+            require(
+                nodeOperatorData.totalSigningKeys >= _decodedCallData[i].stakingLimit,
+                NOT_ENOUGH_SIGNING_KEYS
+            );
+        }
+    }
+
+    function _getNodeOperatorData(
+        uint256 _nodeOperatorId
+    ) private view returns (NodeOperatorData memory _nodeOperatorData) {
+        (
+            ,
```

Comment:
Let's use such an approach for unused variables for clarity: https://github.com/lidofinance/lido-dao/blob/cadffa46a2b8ed6cfa1127fca2468bae1a82d6bf/contracts/0.8.9/StakingRouter.sol#L492